crates/pipeline-manager/src/api/mod.rs

```diff
@@ -458,8 +459,13 @@ pub async fn run(
     let addr = env::var("BANNER_ADDR").unwrap_or(bind_address);
     let url = format!("http://{}:{}", addr, port);
 
-    println!(
-        r"
+    // Lock both out streams so that the banner is printed in one go
+    // and not interrupted by log messages from other threads.
```

Comment:
Clever, I never thought to do this (in Rust or other languages).

Reply:
(although, if more than one piece of code in a program does this, the ordering is important otherwise we could deadlock)